### PR TITLE
Added metadata to line_on_off.py

### PR DESCRIPTION
**Pull request by Azaya89**
Added metadata to examples/plotting/line_on_off.py and did some minor code edits on the file.
- [x] issues: #11765 
- [x] tests added / passed

### CODE REVIEW

File: examples/plotting/line_on_off.py [hunk 1]
```diff
@@ -1,4 +1,11 @@
-""" Example demonstrating turning lines on and off - with JS only
+""" Example demonstrating turning lines on and off - with JS only.
+It involves checking and unchecking the checkboxes representing
+the plotted lines to turn the lines on/off.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure.line, bokeh.layouts.row
```
Comment (tcmetzger) [inline examples/plotting/line_on_off.py]: I'd add CustomJS:
```suggestion
    :apis: bokeh.plotting.figure.line, bokeh.layouts.row, bokeh.models.CustomJS
```

File: examples/plotting/line_on_off.py [hunk 2]
```diff
@@ -1,4 +1,11 @@
...
+    :refs: :ref:`ug_basic_lines_line_multiple`
```
Comment (tcmetzger) [inline examples/plotting/line_on_off.py]: Let's also add JS Callbacks here, and maybe the checkbox, while we are at it:
```suggestion
    :refs: :ref:`ug_interaction_js_callbacks`, :ref:`_ug_interaction_widgets_examples_checkboxgroup`, :ref:`ug_basic_lines_line_multiple`
```

File: examples/plotting/line_on_off.py [hunk 3]
```diff
@@ -1,4 +1,11 @@
...
+    :keywords: line_on_off, checkbox, CustomJS
```
Comment (tcmetzger) [inline examples/plotting/line_on_off.py]: I don't know if line_on_off is a keyword that people would search for - how about we keep it simple and just go with line chart:

```suggestion
    :keywords: line, checkbox, CustomJS
```

**Review by tcmetzger — CHANGES_REQUESTED** [review]
Thank you, just some small requests!

File: examples/plotting/line_on_off.py (hunk 3 above)
Comment (Azaya89) [reply examples/plotting/line_on_off.py]: Executed all the requested changes.

File: examples/plotting/line_on_off.py [hunk 4]
```diff
@@ -1,4 +1,13 @@
-""" Example demonstrating turning lines on and off - with JS only
+""" Example demonstrating turning lines on and off - with JS only.
+It involves checking and unchecking the checkboxes representing
+the plotted lines to turn the lines on/off.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure.line, bokeh.layouts.row, bokeh.models.CustomJS
+    :refs: :ref:`ug_basic_lines_line_multiple`,
+    :ref:`_ug_interaction_widgets_examples_checkboxgroup`,
+    :ref:`ug_interaction_js_callbacks`
```
Comment (bryevdv) [inline examples/plotting/line_on_off.py]: Unfortunately, due to limitations in the sphinx extension, these all need to be on a single line. If the line is too long  for the linter then `# noqa` can be added to the end.

File: examples/plotting/line_on_off.py [hunk 5]
```diff
@@ -19,7 +28,7 @@
 checkbox = CheckboxGroup(labels=["Line 0", "Line 1", "Line 2"],
                          active=[0, 1, 2], width=100)
 
-callback = CustomJS(args=dict(l0=l0, l1=l1, l2=l2, checkbox=checkbox), code="""
+callback = CustomJS(args={"l0":l0, "l1":l1, "l2":l2,"checkbox":checkbox}, code="""
```
Comment (bryevdv) [inline examples/plotting/line_on_off.py]: I prefer to use `dict` for these cases. The names need to be proper identifiers, and using kwargs to `dict` helps reinforce that.

File: examples/plotting/line_on_off.py [hunk 6]
```diff
@@ -10,7 +19,7 @@
 from bokeh.plotting import figure, show
 
 p = figure()
-props = dict(line_width=4, line_alpha=0.7)
+props = {"line_width":4, "line_alpha":0.7}
```
Comment (bryevdv) [inline examples/plotting/line_on_off.py]: I'm -0 on this change (I'd be happy to just revert it) but if we do keep it, then it needs to match stylistically with the rest of the codebase:
```suggestion
props = {"line_width": 4, "line_alpha": 0.7}
```

**Review by bryevdv — CHANGES_REQUESTED** [review]
Thanks @Azaya89 a few comments to address

File: examples/plotting/line_on_off.py (hunk 6 above)
Comment (Azaya89) [reply examples/plotting/line_on_off.py]: All changes executed. I matched the style of the codebase by adding the space.

File: examples/plotting/line_on_off.py [hunk 7]
```diff
@@ -1,5 +1,11 @@
-""" Example demonstrating turning lines on and off - with JS only
-
+""" Example demonstrating turning lines on and off - with JS only.
+It involves checking and unchecking the checkboxes representing
+the plotted lines to turn the lines on/off.
+
+.. bokeh-example-metadata::
+    :apis: bokeh.plotting.figure.line, bokeh.layouts.row, bokeh.models.CustomJS
+    :refs: :ref:`ug_basic_lines_line_multiple`,:ref:`_ug_interaction_widgets_examples_checkboxgroup`,:ref:`ug_interaction_js_callbacks`
```
Comment (bryevdv) [inline examples/plotting/line_on_off.py]: one ref slightly off:
```suggestion
    :refs: :ref:`ug_basic_lines_multi`,:ref:`_ug_interaction_widgets_examples_checkboxgroup`,:ref:`ug_interaction_js_callbacks`
```

**Review by bryevdv — CHANGES_REQUESTED** [review]
Thanks @Azaya89 one last comment about a reference

File: examples/plotting/line_on_off.py (hunk 7 above)
Comment (Azaya89) [reply examples/plotting/line_on_off.py]: Done.

**Review by bryevdv — APPROVED** [review]
Thanks @Azaya89 a few comments to address

**Comment by bryevdv** [comment]
Thank @Azaya89 !

**Pull request merged**